consensus/src/chained_bft/event_processor.rs

```diff
@@ -117,14 +111,17 @@ impl<T: Payload> EventProcessor<T> {
             txn_manager,
             network,
             storage,
-            sync_manager,
             time_service,
             enforce_increasing_timestamps,
             last_vote_sent,
             validators,
         }
     }
 
+    fn create_block_retriever(&self, deadline: Instant, author: Author) -> BlockRetriever {
```

Comment:
Why did you choose to expose block retriever to event processor? Wouldn't it be more modular to have block retriever private to block store?

Reply:
it saves a few typing, otherwise we need to pass `network, deadline, author` into block_store or the function calls.
We can make all the functions of BlockRetriever private.

Reply:
Ok, I might be nitpicking here, but if we're already talking about refactoring this code to make  it cleaner, do you think that BlockRetriever struct is not very clean (one constructor that specifies the parameters for a function and then a function call)?

My intuition is that it'd be nicer if we'd have a struct `SyncContext`, with a field for deadline and a field for preferred peer. `SyncContext` is exposed to EventProcessor. The `sync_to` function would receive this context and could pass it on to the `retrieve_block_for_qc` function (which could then be static, it'd receive a SyncContext and a clone of network).

I guess what I'm talking about is getting closer to the original code :). Don't have very strong opinion here, so I'm fine if you think this PR is actually cleaner, just want to hear your opinion about it.

Reply:
The problem is `block_store` doesn't carry a network like `sync_manager` so we can not only pass `SyncContext` alone and I don't think it's good for it to hold network.

Then it becomes either 
-`SyncContext` carries the network which is essentially a `BlockRetriever`
or
- `sync_to` accepts a `SyncContext` and a `ConsensusNetworkImpl` and construct the `BlockRetriever` inside.

I thought it's better to avoid unnecessary layer of abstraction since BlockRetriever is a simple struct.

Reply:
Let it be
@bors-libra r+